### ISSUE TITLE
Changing into using Swift UI instead of Storyboard

### ISSUE THREAD

**Issue by YazeedAlKhalaf** [flutter/flutter#56668]
## Use case

This is not a problem in the meantime but could be in the near future. The problem is that Flutter uses `storyboard` instead of `Swift UI`. As I said not a real problem but could be in the future, better to consider it as a feature :).

## Proposal

The proposal is to switch from using `storyboard` to `Swift UI`. Short but to the point.

**Comment by YazeedAlKhalaf** [comment]
It will break the API so it may be postponed till Flutter 2.0?

**Comment by YazeedAlKhalaf** [comment]
Any updates?

**Comment by jmagman** [comment]
Surprising, I can't find a duplicate at the moment.

I know this is requesting that the default template be swapped, but in the meantime you're welcome to open your `ios/Runner.xcworkspace` and add SwiftUI interfaces as needed.

**Comment by YazeedAlKhalaf** [comment]
Yup that’s right, but it is a temporary solution. So I hope the flutter team can take a look at this issue. @timsneath

**Comment by timsneath** [comment]
Right now, my understanding is that apps need to use an Xcode storyboard for the app's launch screen: https://developer.apple.com/news/?id=03262020b. Even though Swift UI is clearly their direction, you're going to at least need one storyboard in your app. Swift UI apps include `LaunchScreen.storyboard` themselves. 

So it's not quite clear what the benefit would be here for the time being: it might increase app size while not reducing dependencies. Feel free to comment if you or others have additional insights to offer here.

**Comment by YazeedAlKhalaf** [comment]
I mean changing the `Main.storyboard` to `Main.swift`. I know the launch screen has to be a storyboard, but using swift is better I guess. Correct if I am wrong 🎈

**Comment by jmagman** [comment]
SwiftUI requires a minimum of iOS 13, and Flutter supports iOS 9.  We would need to keep the current Swift creation template to support iOS 9 (yes I know this embeds the Swift runtime). 

We could add a new non-default SwiftUI `flutter create --ios-language swiftui` template (currently supports `objc`, `swift`), which would only support iOS 13 and newer, dropping support for the [7% of users on < iOS 13](https://developer.apple.com/support/app-store/).

![Screen Shot 2021-11-18 at 12 52 24 PM](https://user-images.githubusercontent.com/682784/142494888-bb352b6e-6387-424d-a89d-4f1e3ffc2718.png)

This hasn't been a priority since it increases our testing matrix for not much gain: it drops support for < iOS 13 and most Flutter developers never directly edit the code in their Xcode project.
